### ISSUE TITLE
None of the trajectories converge, please provide a working example

### ISSUE THREAD

**Issue by ignatovmg** [bene837/af2seq#4]
Thanks a lot this work. I've been testing some proteins from the recent [paper](https://www.nature.com/articles/s41586-024-07601-y), such as rhomboid protease, and haven't been able to generate a single scaffold yet by using the example script. Here is my script and the target scaffold pdb file, based on 3ZEB pdb entry (same as what was used in the paper according to Methods): 

Script run.py:
```python
import os
import sys
os.environ["TF_FORCE_UNIFIED_MEMORY"] = "1"
os.environ["XLA_PYTHON_CLIENT_MEM_FRACTION"] = "2.0"

from path import Path
import prody
import numpy as np
import click 

from af2seq.design.utils import generate_start_sequence
from af2seq import GradientDesign


@click.command()
@click.argument('seed', type=int)
@click.argument('outdir', type=str)
@click.argument('trg_pdb', type=str)
@click.option('--enable-sc-loss', is_flag=True)
def cli(seed, outdir, trg_pdb, enable_sc_loss=False):
    # no Cystenine
    abc = ['A', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']

    outdir = Path(outdir).mkdir_p()
    sequence = generate_start_sequence(trg_pdb)
    print(f'Generated start sequence:\n{sequence}')

    fix_resnums = []

    trg_ag = prody.parsePDB(trg_pdb)
    trg_reslist = [r for r in trg_ag.iterResidues()]
    trg_resnums = [x.getResnum().item() for x in trg_reslist]
    
    fix_pos = [trg_resnums.index(x) + 1 for x in fix_resnums]
    sequence = list(sequence)
    for fix_i in fix_pos:
        sequence[fix_i-1] = trg_reslist[fix_i-1].getSequence()[0]
        
    # mutate 10% like in the paper
    unfix_pos = sorted(set(range(1, len(sequence)+1)) - set(fix_pos))
    rng = np.random.default_rng(seed)
    rng.shuffle(unfix_pos)
    mutate_pos = unfix_pos[:len(unfix_pos) // 10]
    mut_ids = rng.integers(0, len(abc)-1, len(mutate_pos))
    for pos, aa_id in zip(mutate_pos, mut_ids):
        sequence[pos-1] = abc[aa_id]
    
    sequence = ''.join(sequence)
    print('Fixed residues: ' + str(fix_pos))
    print(f'Fixed start sequence:\n{sequence}')
    print(f'Output dir: {outdir}')

    d = GradientDesign(
        datadir='../../../af2seq', 
        output_path=outdir
    )

    # Set the weights on the loss function
    loss = {"fape": 1.0, "ptm": 0.2, "plddt": 0.2}

    # Design settings
    d.design(trg_pdb,
            start_seq=sequence,
            iterations=500,
            lr=1e-3,             # Learning rate to start with (uses ADAM)
            chains = None,       # Chains we want to design.
            aa_mask=['C'],       # add here which amino acids you wish to mask for updating
            fix_pos=fix_pos,     # Position mask, give the index of amino acids of the target not to update (only works for ptm model for now!)
            recycles=0,          # number of recycles
            clampval=0.0,        # weight of the clamping of the FAPE loss.
            modeltype='ptm',
            enable_sc_loss=fix_pos if enable_sc_loss else None,
    #mode = 'gd',
            loss = loss)

if __name__ == '__main__':
    cli()
```

To run:
```bash
python run_af2seq.py 123456 output/ 3zeb_prot_only.pdb
```

I ran this 30 times with different seeds and the best RMSD I achieved was 4.2A (which is huge for a small protein) and the best TM-score was 0.27. Pdb file I'm using: 
[3zeb_prot_only.pdb.zip](https://github.com/user-attachments/files/16313646/3zeb_prot_only.pdb.zip)

Perhaps I'm doing something wrong. Would be great if you could provide a working example of a starting sequence, which converges to the target scaffold.

**Comment by casperg92** [comment]
Hi ignatov,

Two things that will make it work. 1) we used secondary structure element initializiation for fast convergence, i would use pydssp to generate a starting sequence. And secondly, for 3ZEB you should use the distogram loss on top of the FAPE loss. The settings I used for our 3ZEB runs where:

loss = {"fape": 0.5, "distogram": 1.0, "ptm": 0.1, "plddt": 0.0}

Let me know if this works for you!

**Comment by ignatovmg** [comment]
Thanks for your response. As you can see from my code, pydssp is already used here to generate the starting sequence: `sequence = generate_start_sequence(trg_pdb)`. 

I will try your loss reweighting scheme, and will let you know if that helps.

**Comment by ignatovmg** [comment]
Update: I ran 20 more trajectories with `loss = {"fape": 0.5, "distogram": 1.0, "ptm": 0.1, "plddt": 0.0}` and none of them converged - best reported tm score is 0.27. Do you have an idea what else might help?
